File: fabric/src/main/java/me/pandamods/pandalib/fabric/platform/GameHelperImpl.java
/*
 * Copyright (C) 2024 Oliver Froberg (The Panda Oliver)
 *
 * This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  any later version.
 *
 * You should have received a copy of the GNU Lesser General Public License
 *  along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package me.pandamods.pandalib.fabric.platform;

import dev.architectury.utils.Env;
import me.pandamods.pandalib.platform.services.GameHelper;
import net.fabricmc.api.EnvType;
import net.fabricmc.loader.api.FabricLoader;

import java.nio.file.Path;

public class GameHelperImpl implements GameHelper {
	@Override
	public boolean isDevelopmentEnvironment() {
		return FabricLoader.getInstance().isDevelopmentEnvironment();
	}

	@Override
	public boolean isProductionEnvironment() {
		return !FabricLoader.getInstance().isDevelopmentEnvironment();
	}

	@Override
	public Env getEnvironment() {
		return switch (FabricLoader.getInstance().getEnvironmentType()) {
			case CLIENT -> Env.CLIENT;
			case SERVER -> Env.SERVER;
		};
	}

	@Override
	public boolean isClient() {
		return FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT;
	}

	@Override
	public boolean isServer() {
		return FabricLoader.getInstance().getEnvironmentType() == EnvType.SERVER;
	}

	@Override
	public Path getGameDir() {
		return FabricLoader.getInstance().getGameDir().toAbsolutePath().normalize();
	}

	@Override
	public Path getConfigDir() {
		return FabricLoader.getInstance().getConfigDir().toAbsolutePath().normalize();
	}

	@Override
	public Path getModDir() {
		return getGameDir().resolve("mods");
	}
}

File: fabric/src/main/kotlin/platform/GameHelperImpl.kt
/*
 * Copyright (c) 2025. Oliver Froberg
 *
 * This code is licensed under the GNU Lesser General Public License v3.0
 * See: https://www.gnu.org/licenses/lgpl-3.0-standalone.html
 */

package dev.pandasystems.pandalib.fabric.platform

import com.google.auto.service.AutoService
import dev.pandasystems.pandalib.core.platform.GameHelper
import dev.pandasystems.pandalib.fabric.PandaLibFabric
import dev.pandasystems.pandalib.api.utils.Environment
import net.fabricmc.api.EnvType
import net.fabricmc.loader.api.FabricLoader
import net.minecraft.server.MinecraftServer
import java.nio.file.Path

@AutoService(GameHelper::class)
class GameHelperImpl : GameHelper {
	override val isDevelopment = FabricLoader.getInstance().isDevelopmentEnvironment
	override val isProduction = !FabricLoader.getInstance().isDevelopmentEnvironment

	override val environment = when (FabricLoader.getInstance().environmentType) {
		EnvType.CLIENT -> Environment.CLIENT
		EnvType.SERVER -> Environment.DEDICATED_SERVER
	}
	override val isClient = FabricLoader.getInstance().environmentType == EnvType.CLIENT
	override val isDedicatedServer = FabricLoader.getInstance().environmentType == EnvType.SERVER

	override val gameDir: Path = FabricLoader.getInstance().gameDir.toAbsolutePath().normalize()
	override val configDir: Path = FabricLoader.getInstance().configDir.toAbsolutePath().normalize()
	override val modDir: Path = gameDir.resolve("mods")

	override val server: MinecraftServer?
		get() = PandaLibFabric.server
}
